libraries/plugins/chain/chain_plugin.cpp

```diff
@@ -432,16 +420,27 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
 void chain_plugin::plugin_startup()
 {
    ilog( "Starting chain with shared_file_size: ${n} bytes", ("n", my->shared_memory_size) );
+
    chain_id_type chain_id = my->genesis.compute_chain_id();
 
-   my->start_write_processing();
-   if(my->shared_memory_dir.generic_string() == "")
+   if(my->shared_memory_dir.generic_string() == "") {
       my->shared_memory_dir = app().data_dir() / chain_id.str() / "blockchain";
+   }
+
+   //correct directories
+   if( ! my->genesis.is_private_net && bfs::exists( app().data_dir() / "blockchain" ) ){
+      bfs::create_directories ( my->shared_memory_dir );
```

Comment:
I would delete "! my->genesis.is_private_net &&" from this condition. But its not a must :)

Reply:
I was assuming that sophiatx_app_data/blockchain is used for default, i.e. mainnet net.

Reply:
but according to changes, this condition is always entered(does not matter if it is private net or not):
if(my->shared_memory_dir.generic_string() == "") {
      my->shared_memory_dir = app().data_dir() / chain_id.str() / "blockchain";
   }

so even mainnet is going to use sophiatx_app_data/chain_id/blockchain directory

Reply:
Yes, for private nets, the appropriate directory is created, but only for mainnet the old data are migrated from sophiatx_app_data/blockchain to the new directory

Reply:
ok, get your point now :)